File: CourseProjects/CourseProjects/JavaEclipseProjects/JavaLanguageExercises/src/inheritance/finish/Student.java
package inheritance.finish;

public class Student {
	public Student(String name, String subject, int yearOfStudy, double[] marks) {
		this.subject = subject;
		this.marks = marks;
		this.name = name;
		this.yearOfStudy = yearOfStudy;
	}
	public double average() {
		double total = 0;
		for(int i=0;i<marks.length;i++) {
			total += marks[i];
		}
		return total / marks.length;
	}
	public String toString() {
		StringBuffer msg = new StringBuffer();
		msg.append(name);
		msg.append(" studying ");
		msg.append(subject);
		msg.append(" in year ");
		msg.append(yearOfStudy);
		msg.append(" with results");
		for(int i=0;i<marks.length;i++) {
			msg.append(" ");
			msg.append(marks[i]);
		}
		return msg.toString();
	}
	private String name;
	private String subject;
	private int yearOfStudy;
	private double [] marks;
}

File: CourseProjects/CourseProjects/VisualStudioProjects/Languages/LanguageExercisesCS/InheritanceStart/Student.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace InheritanceStart {
    internal class Student {
        public Student(string name, string subject, int year, double[] marks) {
            this.name = name;
            this.subject = subject;
            this.year = year;
            this.marks = marks;
        }

        public int Year {
            set { year = value; }
        }

        public double Average() {
            double total = 0;
            foreach (double mark in marks) {
                total += mark;
            }
            return total/marks.Length;
        }

        public override string ToString() {
            string msg = "{0} studying {1} in year {2}";
            return string.Format(msg, name, subject, year);
        }

        private string name;
        private string subject;
        private int year;
        private double[] marks;
    }
}